### ISSUE TITLE
Documentation issue regarding MDAnalysis.lib.distances.calc_bonds 

### ISSUE THREAD

**Issue by xiki-tempula** [MDAnalysis/mdanalysis#2004]
The documentation for MDAnalysis.lib.distances.calc_bonds states that 

`MDAnalysis.lib.distances.calc_bonds( *atom1* ,  *atom2* [, box[, result[, backend]]])`

it takes in two atoms which is actually the corrdinate of two atoms.

The explanation is ok though.

Calculate all distances between a pair of atoms.  *atom1* and *atom2* are both
arrays of coordinates, where atom1[i] and atom2[i] represent a bond.

In comparison to distance_array and self_distance_array which calculate distances
between all combinations of coordinates, calc_bonds can be used to calculate distance
between pairs of objects, similar to::

       numpy.linalg.norm(a - b) for a, b in zip(coords1, coords2)

The optional argument *box* applies minimum image convention if supplied.
*box* can be either orthogonal or triclinic

If a 1D numpy array of dtype ``numpy.float64`` with ``len(atom1)`` elements is
provided in *result* then this preallocated array is filled. This can speed
up calculations.

*bondlengths = calc_bonds(coords1, coords2 [, box [,result=bondlengths]])*

    Parameters
    ----------
    coords1 : array
        An array of coordinates for one half of the bond.
    coords2 : array
        An array of coordinates for the other half of bond
    box : array
        The unitcell dimesions for this system.
        The dimensions must be provided in the same format as returned
        by :attr:`MDAnalysis.coordinates.base.Timestep.dimensions`: ``[lx,
        ly, lz, alpha, beta, gamma]``.
    result : array, optional
        Preallocated result array which must be same length as coord
        arrays and ``dtype=numpy.float64``. Avoids creating the
        array which saves time when the function is called repeatedly. [None]
    backend
        Select the type of acceleration; "serial" is always available. Other
        possibilities are "OpenMP" (OpenMP).

I think it will be nice if we can have some consistency between the parameters in the documemntation and the parameters in the code.

**Comment by richardjgowers** [comment]
Yeah it looks like the docstring should be fixed to make it clearer

**Comment by zemanj** [comment]
Turns out there are a lot more issues with the docs of ``MDAnalysis.lib.distances``. This will take a while...

**Comment by kain88-de** [comment]
@zemanj  can you make a list with functions that need work on the docstring. We need can then work on this one by one. Smaller PRs result in quicker review.
